dist/index.js

```diff
@@ -37971,18 +37971,43 @@ function findSuccessfulCommit(workflow_id, run_id, owner, repo, branch, lastSucc
             process.stdout.write("\n");
             process.stdout.write(`Workflow Id not provided. Using workflow '${workflow_id}'\n`);
         }
-        // fetch all workflow runs on a given repo/branch/workflow with push and success
-        const shas = yield octokit
-            .request(`GET /repos/${owner}/${repo}/actions/workflows/${workflow_id}/runs`, {
-            owner,
-            repo,
-            // on non-push workflow runs we do not have branch property
-            branch: lastSuccessfulEvent !== "push" ? undefined : branch,
-            workflow_id,
-            event: lastSuccessfulEvent,
-            status: "success",
-        })
-            .then(({ data: { workflow_runs } }) => workflow_runs.map((run) => run.head_sha));
+        let shas = [];
+        // if there are several events separated by comma
+        if (lastSuccessfulEvent.includes(",")) {
+            // find the greatest workflow id among the types and retrieve the sha
+            const events = lastSuccessfulEvent.split(",");
+            const workflowIdMap = new Map();
+            for (const event of events) {
```

Comment:
This condition and the separate code path below seems unnecessary. A string with no commas, e.g. "foo", will `split(",")` to `["foo"]` and the iteration will work fine over that single element.